core/main/client/browser.js

```diff
@@ -646,6 +646,27 @@ beef.browser = {
         return this.isS4() || this.isS5() || this.isS6() || this.isS7() || this.isS8();
     },
 
+    /**
+     * Returns true if Webkit based
+     */
+    isWebKitBased: function () {
+        // Currently, [2018-01-08], safari checks isS* provide evidence of webkit
+        // to avoid duplicating code and hence bugs, use the existing checks for Safari.
+        return this.isS4() || this.isS5() || this.isS6() || this.isS7() || this.isS8();
```

Comment:
`isWebKitBased` should not depend on `isS*()` even if it means code duplication.